ec2/ec2.go

```diff
@@ -787,15 +777,35 @@ type IPPerm struct {
 	Protocol     string              `xml:"ipProtocol"`
 	FromPort     int                 `xml:"fromPort"`
 	ToPort       int                 `xml:"toPort"`
-	SourceIPs    []string            `xml:"ipRanges>item>cidrIp"`
```

Comment:
If you have a look at ec2t_test.go - TestIPPerms there is a case where this is exercised - starting from line 491. We set 2 IPPerm rules - 1 with two source groups (both by Id and Name) and source IP 127.0.0.0/24, 1 with just a source IP 200.1.1.34/32. Then, when we list the g0 security group the response from Amazon contains 3 IPPerm rules: 1 for each source group (by Id or Name) and 1 with the two source IPs. The way IPPerm.SourceIPs were defined before - `xml:"ipRanges>item>cidrIp"`, given this XML:
```xml
...
<ipPermissions>
  <item>
    ...
   <ipRanges>
     <item><cidrIp>127.0.0.0/24</cidrIp></item>
     <item><cidrIp>200.1.1.34/32</cidrIp></item>
  </ipRanges>
  ...
```
Rather than SourceIPs: []string{"127.0.0.0/24", "200.1.1.34/32"} you'd get only the first one, because the expected XML format is `<ipRanges><item><cidrIp>127.0.0.0/24</cidrIp><cidrIp>200.1.1.34/32</cidrIp></item></ipRanges>` (and the ec2test server in fact returns this, but the live Amazon servers return a response with the format described above).

The easiest way to verify this is to set debug=true in ec2.go and run go test -gocheck.v -gocheck.f TestIPPerms on launchpad.net/goamz/ec2 - with and without the -amazon flag - the latter fails and you can see the response XML is different from Amazon and that response is not deserialized correctly.